packages/xml-views-quick-fix/src/quick-fix-id.ts

```diff
@@ -0,0 +1,107 @@
+import { find, some } from "lodash";
+import { astPositionAtOffset } from "@xml-tools/ast-position";
+import {
+  XMLDocument,
+  accept,
+  XMLAstVisitor,
+  XMLAttribute,
+  XMLElement,
+} from "@xml-tools/ast";
+import { OffsetRange } from "@ui5-language-assistant/logic-utils";
+
+const ID_PATTERN = /^_IDGen(.+)([1-9]\d*)$/;
+const ID_PREFIX_PATTERN = "_IDGen";
+
+export type QuickFixStableIdInfo = {
+  suggestion: string;
+  offsetRange: OffsetRange;
+};
+
+export function computeQuickFixStableIdInfo(
+  xmlDoc: XMLDocument,
+  errorOffset: OffsetRange
+): QuickFixStableIdInfo | undefined {
+  const biggestIdsCollector = new IdsByElementNameCollectorVisitor();
+  accept(xmlDoc, biggestIdsCollector);
+  const biggestIdsOfElements = biggestIdsCollector.biggestIdsOfElements;
+  const astNode = astPositionAtOffset(xmlDoc, errorOffset.start);
+  if (astNode === undefined || astNode.kind !== "XMLElementOpenName") {
+    return undefined;
+  }
+
+  const xmlElement = astNode.astNode;
+  /* istanbul ignore if - ast node of kind "XMLElementOpenName" will always have name  */
+  if (xmlElement.name === null || xmlElement.syntax.openName === undefined) {
+    return undefined;
+  }
+
+  const isExistIdKey = some(
+    xmlElement.attributes,
+    (attrib) => attrib.key === "id"
+  );
+
+  const suggestion = getQuickFixIdSuggestion(
+    biggestIdsOfElements,
+    xmlElement.name,
+    isExistIdKey
+  );
+
+  const offsetRange = getQuickFixIdRange(xmlElement);
```

Comment:
could we be more specific here? is this the **replace** range?
It may help someone who is unfamiliar with this code understand flow better